### PR TITLE
update more spots that use MP.name

### PR DESCRIPTION
**Pull request by timmysilv**
Same as #5071 but in other places.

I'm a little worried that maybe we shouldn't deprecate `MeasurementProcess.name` at all... seems like we kinda depend on it? My first thought was to just make them more correct (eg. "State", "Probs"), but I don't think any devices have those saved in their `dev.observables` so I suppose this might still be the best way forward

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -83,6 +83,7 @@
   values that are no longer needed.
   [(#5047)](https://github.com/PennyLaneAI/pennylane/pull/5047)
   [(#5071)](https://github.com/PennyLaneAI/pennylane/pull/5071)
+  [(#)](https://github.com/PennyLaneAI/pennylane/pull/)
```
Comment (timmysilv) [inline doc/releases/changelog-dev.md:86]:
```suggestion
  [(#5076)](https://github.com/PennyLaneAI/pennylane/pull/5076)
```

**Comment by josh146** [comment]
@timmysilv I could provide some context if needed since (I think??) I originally added this. The original motivation was mostly `repr`/printing for end users (so that circuits would show as `qml.expval(qml.PauliZ)` when displayed etc.)

**Comment by timmysilv** [comment]
@josh146 we're deprecating `MeasurementProcess.name` and `MeasurementProcess.data`, because they were _always_ "Identity" and `[]`, respectively (see [file at time of adding](https://github.com/PennyLaneAI/pennylane/blob/b32b702e3922345cd524854782e932e64f5aa939/pennylane/measurements/measurements.py#L155-L166)). Every time we checked `MP.name`, it was "Identity", even today:
```pycon
>>> qml.state().name
/Users/matthews/src/github.com/PennyLaneAI/pennylane/pennylane/measurements/measurements.py:196: PennyLaneDeprecationWarning: MeasurementProcess.name is deprecated, and will be removed in an upcoming release. To get the name of an observable from a measurement, use MeasurementProcess.obs.name instead
  warn(
'Identity'
```

that said, `Device` clearly does check it and handle it, even though it was always "Identity". While I hoped this wouldn't affect things, it still can sometimes as we're seeing.

**Comment by josh146** [comment]
got it, thanks @timmysilv!

**Review by lillian542 — APPROVED** [review]

**Review by albi3ro — APPROVED** [review]

**Pull request merged**